pkg/controller/operators/catalog/operator.go

```diff
@@ -437,55 +441,40 @@ func (o *Operator) ExecutePlan(plan *v1alpha1.InstallPlan) error {
 	return nil
 }
 
-func (o *Operator) getSourcesSnapshot() map[registry.SourceKey]registry.Source {
+func (o *Operator) getSourcesSnapshot(includedNamespaces map[string]bool) map[registry.SourceKey]registry.Source {
 	o.sourcesLock.RLock()
 	defer o.sourcesLock.RUnlock()
 	sourcesSnapshot := make(map[registry.SourceKey]registry.Source)
 	for key, source := range o.sources {
-		sourcesSnapshot[key] = source
+		// Only copy catalog sources in included namespaces
+		if _, ok := includedNamespaces[key.Namespace]; ok {
+			sourcesSnapshot[key] = source
+		}
 	}
 
 	return sourcesSnapshot
 }
 
-func (o *Operator) getPreferredSourceKey(plan *v1alpha1.InstallPlan) registry.SourceKey {
-	// Attempt to get the owning subscription
-	ownerRefs := plan.ObjectMeta.GetOwnerReferences()
-	var ownerSub metav1.OwnerReference
-
-	for _, ref := range ownerRefs {
-		if ref.Kind == subscriptionv1alpha1.SubscriptionKind {
-			ownerSub = ref
-			break
-		}
-	}
-
-	// Form the subscription key (owning subcription must be in the same namespace as the install plan)
-	subKey := registry.SubscriptionKey{Name: ownerSub.Name, Namespace: plan.Namespace}
-
-	o.subscriptionsLock.RLock()
-	sub, ok := o.subscriptions[subKey]
-	o.subscriptionsLock.RUnlock()
-
-	var preferredSourceKey registry.SourceKey
-
+func (o *Operator) getPreferredSourceKey(plan *v1alpha1.InstallPlan, includedNamespaces map[string]bool) registry.SourceKey {
```

Comment:
I think this method makes more sense if it takes `sourcesSnapshot` as an arg instead of included namespaces, since you've already filtered the sources by includedNamespaces above